### ISSUE TITLE
[SM] stack[0].getFileName() missing

### ISSUE THREAD

**Issue by stefanpenner** [jxcore/jxcore#728]
The V8 specific (I believe) `stack[0].getFileName()` method is missing in SM (and likely vms), some node code depends on this, to discover the location (on-disk) of a caller.

If there is an alternative for SM, I could just implement it that way. Although it would be nice if some standard was eventually derived.

**Comment by obastemur** [comment]
`JXcore` SM build has a patch for V8's `Error.captureStackTrace` and it should behave similarly.

```
try {
  throw new Error("AAA");
} catch(e) {
  Error.captureStackTrace(e);
  console.log(e.stack[0].getFileName());
}
```

I've just tested the code above and it behaves as expected with SM build.

**Comment by stefanpenner** [comment]
Although I get get it to work in isolation, i appear to not be able to get it to work in my specific (more complex example). I'm going to try and isolate it further

**Comment by stefanpenner** [comment]
Actually, I believe my example is more similar to this SO response: http://stackoverflow.com/a/13227808

**Comment by obastemur** [comment]
@stefanpenner Thanks for all the details. Fix will be part of 0.3.1.0 (https://github.com/jxcore/jxcore/commit/ee19070c0fab093e91c119cb1a4d7d8277359352)

**Comment by stefanpenner** [comment]
@obastemur very nice, will give 0.3.1.0 a go sometime.

Keep up the good work!
